### PR TITLE
Move flexiMenu functions to CfgFunctions

### PR DESCRIPTION
**Pull request by dedmen**
tracy_2019-02-08_14-30-35

This triggers me.

### CODE REVIEW

**Comment by commy2** [comment]
Could delete these: https://github.com/CBATeam/CBA_A3/blob/master/addons/ui/script_component.hpp#L83-L86

However, before the functions are named `cba_flexiMenu_fnc_execute`, but now they are named ` `cba_fnc_flexiMenu_execute` etc, or am I missing something?

**Comment by dedmen** [comment]
Might be right yeah. But then we previously had both 
https://github.com/CBATeam/CBA_A3/pull/1054/files#diff-d0bf4bd8fc69fef6ce8641586ab5da29L23
cba_flexiMenu_fnc_add and 
https://github.com/CBATeam/CBA_A3/pull/1054/files#diff-1ed3566cd26e88a81c404dde0895265bR4
cba_fnc_flexuMenu_add

I guess I'll just move them to PREP then?

**Comment by commy2** [comment]
master:
`cba_ui_fnc_execute`

PR:
`cba_fnc_flexiMenu_execute`

File: addons/ui/flexiMenu/init.sqf [hunk 1]
```diff
@@ -8,22 +8,6 @@ GVAR(target) = objNull; // current object for interaction
...
-PREP_SUB(flexiMenu,menuShortcut);
-PREP_SUB(flexiMenu,mouseButtonDown);
-PREP_SUB(flexiMenu,highlightCaretKey);
-PREP_SUB(flexiMenu,execute);
-
-// TODO: move these to public CfgFunctions soon.
-PREP_SUB(flexiMenu,add);
-PREP_SUB(flexiMenu,remove);
-PREP_SUB(flexiMenu,setObjectMenuSource);
-
```
Comment (commy2) [inline addons/ui/flexiMenu/init.sqf]:
```suggestion
FUNC(keyDown) = CBA_fnc_flexiMenu_keyDown;
FUNC(keyUp) = CBA_fnc_flexiMenu_keyUp;
FUNC(menu) = CBA_fnc_flexiMenu_menu;
FUNC(list) = CBA_fnc_flexiMenu_list;
FUNC(getMenuDef) = CBA_fnc_flexiMenu_getMenuDef;
FUNC(getMenuOption) = CBA_fnc_flexiMenu_getMenuOption;
FUNC(menuShortcut) = CBA_fnc_flexiMenu_menuShortcut;
FUNC(mouseButtonDown) = CBA_fnc_flexiMenu_mouseButtonDown;
FUNC(highlightCaretKey) = CBA_fnc_flexiMenu_highlightCaretKey;
FUNC(execute) = CBA_fnc_flexiMenu_execute;
FUNC(add) = CBA_fnc_flexiMenu_add;
FUNC(remove) = CBA_fnc_flexiMenu_remove;
FUNC(setObjectMenuSource) = CBA_fnc_setObjectMenuSource;
```

tbh, this whole subcomponent is cancer.

Comment (dedmen) [reply addons/ui/flexiMenu/init.sqf]: Are you sure about this? could just move the old PREP_SUB that was there to PREP.hpp and get same effect as on master branch

Comment (commy2) [reply addons/ui/flexiMenu/init.sqf]: PREP_SUB is there because the filepath is different. The files are in a sub folder.

Comment (dedmen) [reply addons/ui/flexiMenu/init.sqf]: I know. But does anything speak against just using the old PREP_SUB code in PREP.hpp? Instead of just copying the functions to a new/old name?

Comment (commy2) [reply addons/ui/flexiMenu/init.sqf]: Idk how PREP_SUB works. I just assume it recompiles every time and is nothing like PREP at all.

Comment (dedmen) [reply addons/ui/flexiMenu/init.sqf]: Correct. Direct compile preproc..

File: addons/ui/flexiMenu/init.sqf [hunk 2]
```diff
@@ -32,4 +16,18 @@ GVAR(lastAccessCheck) = [0, -1];
...
+FUNC(list) = CBA_fnc_flexiMenu_list;
+FUNC(getMenuDef) = CBA_fnc_flexiMenu_getMenuDef;
+FUNC(getMenuOption) = CBA_fnc_flexiMenu_getMenuOption;
+FUNC(menuShortcut) = CBA_fnc_flexiMenu_menuShortcut;
+FUNC(mouseButtonDown) = CBA_fnc_flexiMenu_mouseButtonDown;
+FUNC(highlightCaretKey) = CBA_fnc_flexiMenu_highlightCaretKey;
+FUNC(execute) = CBA_fnc_flexiMenu_execute;
+FUNC(add) = CBA_fnc_flexiMenu_add;
+FUNC(remove) = CBA_fnc_flexiMenu_remove;
+FUNC(setObjectMenuSource) = CBA_fnc_setObjectMenuSource;
```
Comment (commy2) [inline addons/ui/flexiMenu/init.sqf]: Why did you move these down here? Now L32/L16f add a function to eventhandler that doesn't exist yet.

**Comment by commy2** [comment]
Can we also delete these silly `description` entries from the other functions? They do nothing.

File: addons/ui/flexiMenu/init.sqf (hunk 2 above)
Comment (dedmen) [reply addons/ui/flexiMenu/init.sqf]: Because I wanted them all neatly together :D 
I'll move them to the top tomorrow if I remember.

File: addons/ui/flexiMenu/init.sqf [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
```
Comment (commy2) [inline addons/ui/flexiMenu/init.sqf]: Now you moved them above the `isDedicated exitWith` thingy.

Comment (dedmen) [reply addons/ui/flexiMenu/init.sqf]: Yeah. Because what happens if people call the funcs on server? Optimally they should do nothing I guess

Comment (dedmen) [reply addons/ui/flexiMenu/init.sqf]: Leave me alone I don't have kids!

**Comment by dedmen** [comment]
wat?
https://github.com/CBATeam/CBA_A3/pull/1054/files#diff-d0bf4bd8fc69fef6ce8641586ab5da29R17
It's right there.
And right there https://github.com/CBATeam/CBA_A3/pull/1054/files#diff-1ed3566cd26e88a81c404dde0895265bR12
so what?...


should
https://github.com/CBATeam/CBA_A3/pull/1054/files#diff-d0bf4bd8fc69fef6ce8641586ab5da29R17
be
`FUNC(setObjectMenuSource) = CBA_fnc_flexiMenu_setObjectMenuSource;` ?

**Comment by commy2** [comment]
idk, dude, it's the flexi menu dude

**Comment by commy2** [comment]
What about not doing the = stuff and instead to make use of CfgFunctions with a different tag?
https://community.bistudio.com/wiki/Arma_3_Functions_Library#Tag

**Comment by dedmen** [comment]
We need some of them twice anyway as they were defined in CfgFunctions and as a PREP with different names.

**Comment by commy2** [comment]
What I don't want is to introduce new (duplicate) globals with this PR if it can be avoided.
